lib/ABI/FunctionType.cpp

```diff
@@ -95,6 +98,13 @@ static model::QualifiedType getTypeOrDefault(const model::QualifiedType &Type,
     return buildType(Register, Binary);
 }
 
+struct InternalLayout {
+  std::decay<decltype(Layout::Arguments)>::type Arguments;
+  std::decay<decltype(Layout::ReturnValue)>::type ReturnValue;
+  std::decay<decltype(Layout::CalleeSavedRegisters)>::type CalleeSavedRegisters;
+  std::decay<decltype(Layout::FinalStackOffset)>::type FinalStackOffset;
+};
```

Comment:
Why do we need this instead of just using `Layout`?

Reply:
Because `Layout` is immutable ;)

Reply:
So if no longer necessary we can elide it?

Reply:
It's already so in my working branch.